bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/ExecutorImpl.scala

```diff
@@ -13,39 +16,56 @@ import scalaz.concurrent.Task
 /** An executor that maintains state in a pair of `TaskRef`s. */
 class ExecutorImpl private (cancelRef: TaskRef[Set[SPObservationID]], stateRef: TaskRef[Map[SPObservationID, Executor.ExecState]]) {
 
+  private var loc = new Peer("localhost", 8443, null)
+
+  def host(): Peer = loc
+  def host(l: Peer): Unit = { loc = l }
+
   private def recordState(id: SPObservationID)(s: ExecState): Task[Unit] =
     stateRef.modify(_ + (id -> s))
 
   private def go(id: SPObservationID): Task[Boolean] = 
     cancelRef.get.map(!_(id))
 
-  def sequence(sequenceConfig: ConfigSequence): (Set[System], List[Step.Step]) =
-    sequenceConfig.getAllSteps.toList.map(Step.step).unzip match {
-      case (s, l) => (s.suml, l)
-    }
+  def read(oid: SPObservationID): SeqexecFailure \/ ConfigSequence =
+      SeqExecService.client(loc).sequence(oid).leftMap(SeqexecFailure.ODBSeqError(_))
+
+  def sequence(sequenceConfig: ConfigSequence):  SeqexecFailure \/ (Set[System], List[Step.Step]) = {
+    val a = sequenceConfig.getAllSteps.toList.map(Step.step).sequenceU
+    a map { _.unzip match {
+      case (l, s) => (l.suml, s)
+    }}
+  }
 
   // todo: it is an error to run a sequence with an existing ExecState != s
   private def runSeq(id: SPObservationID, s: ExecState): Task[(ExecState, NonEmptyList[SeqexecFailure] \/ Unit)] =
     cancelRef.modify(_ - id) *> recordState(id)(s) *> run(go(id), recordState(id))(s)
 
-  def start(id: SPObservationID, sequenceConfig: ConfigSequence): Task[(ExecState, NonEmptyList[SeqexecFailure] \/ Unit)] = {
-    sequence(sequenceConfig) match {
+  def start(id: SPObservationID): SeqexecFailure \/ Unit = for {
```

Comment:
The return type tells me that I either get an error or I get nothing. I'd expect, in the non-case error, to get something I can use to follow or interrupt the sequence, or do we rely on the id only?